tiledb/sm/array/array.cc

```diff
@@ -620,25 +586,25 @@ std::vector<shared_ptr<const Enumeration>> Array::get_enumerations(
       // Create a vector of paths to be loaded.
       std::vector<std::string> paths_to_load;
       for (auto& enmr_name : enmrs_to_load) {
-        auto path = array_schema_latest_->get_enumeration_path_name(enmr_name);
+        auto path = array_schema_latest().get_enumeration_path_name(enmr_name);
         paths_to_load.push_back(path);
       }
 
       // Load the enumerations from storage
-      loaded = array_dir_.load_enumerations_from_paths(
-          paths_to_load, get_encryption_key(), memory_tracker_);
+      loaded = array_directory().load_enumerations_from_paths(
+          paths_to_load, *encryption_key(), memory_tracker_);
     }
 
     // Store the loaded enumerations in the schema
     for (auto& enmr : loaded) {
-      array_schema_latest_->store_enumeration(enmr);
+      opened_array_->array_schema_latest_ptr()->store_enumeration(enmr);
```

Comment:
I'm guessing this is old? Or am I missing a reason this isn't using the `array_schema_latest().store_enumeration(enmr)` pattern instead?

Reply:
This is because array_schema_latest() is const and store modifies the data.

Reply:
Ah, fair point.